pandas/_libs/tslib.pyx

```diff
@@ -455,18 +455,18 @@ cpdef array_to_datetime(
         set out_tzoffset_vals = set()
         tzinfo tz_out = None
         bint found_tz = False, found_naive = False
-        cnp.broadcast mi
+        cnp.flatiter it = cnp.PyArray_IterNew(values)
```

Comment:
I think PyArray_IterNew is the old function and we should be using NpyIter_New instead - does Cython offer that?

https://numpy.org/doc/stable/reference/c-api/iterator.html#converting-from-previous-numpy-iterators

Reply:
It'd doesn't seem to be exposed, there's an open issue :( . 

https://github.com/numpy/numpy/issues/21014.